transaction/std/src/memo/mod.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2018-2021 The MobileCoin Foundation
+
+//! Defines an object for each known high-level memo type,
+//! and an enum to allow matching recovered memos to one of these types.
+//!
+//! To add a new memo type, add a new module for it, add a structure,
+//! and make it convertible to MemoPayload.
+//! Then also add it to the MemoType enum and extend the TryFrom logic.
```

Comment:
How is `MemoType` being used?
What happens if a third party wants to add some proprietary memo type for their own internal use?

Reply:
this comment is stale, i need to fix it.

and your questions are good, I will make some adjustments and speak to that in the next commit

Reply:
i added more comments and made all these macros exportable, lmk if this makes all sense to you

Reply:
Yup, thanks!